File: src/snippets/Condition tree/Kotlin/condTree.kt
fun main() {
	var a = (1..4).random()
	var b = (1..4).random()
	lateinit var msg: String
	println("a = $a, b = $b")

	when {
		a > b -> {
			msg = "a > b"
			a++
		}
		a < b -> {
			msg = "a < b"
			b++
		}
		else -> {
			msg = "a == b"
			a++
			b++
		}
	}
	println("$msg\na = $a, b = $b")
}

File: src/snippets/Condition tree/C++/condTree.cpp
#include <iostream>
#include <string>

void print(const int a, const int b) {
	std::cout << "a = " << a << ", b = " << b << '\n';
}
void compare(int a, int b) {
	std::string msg;
	print(a, b);

	if(a > b) {
		msg = "a > b";
		a++;
	} else if(a < b) {
		msg = "a < b";
		b++;
	} else {
		msg = "a == b";
		a++;
		b++;
	}

	std::cout << msg << '\n';
	print(a, b);
	std::cout << '\n';
}

int main() {
	for(int a = 1; a <= 3; a++)
		for(int b = 1; b <= 3; b++)
			compare(a, b);
	return 0;
}
